public/app/features/dashboard/components/ShareModal/SharePublicDashboard.tsx

```diff
@@ -15,25 +16,30 @@ import { ShareModalTabProps } from './types';
 
 interface Props extends ShareModalTabProps {}
 
-// 1. write test for dashboardCanBePublic
-// 2. figure out how to disable the switch
-
 export const SharePublicDashboard = (props: Props) => {
-  const [publicDashboardConfig, setPublicDashboardConfig] = useState<PublicDashboardConfig>({ isPublic: false });
   const dashboardUid = props.dashboard.uid;
+  const orgId = contextSrv.user.orgId;
+  const [publicDashboardConfig, setPublicDashboardConfig] = useState<PublicDashboardConfig>({
+    isPublic: false,
+    publicDashboard: { uid: '', dashboardUid, orgId },
+  });
 
   useEffect(() => {
     getPublicDashboardConfig(dashboardUid)
       .then((pdc: PublicDashboardConfig) => {
+        // empty uid means there isn't a public dashboard entry yet
+        if (pdc.publicDashboard.dashboardUid === '') {
+          pdc.publicDashboard.dashboardUid = dashboardUid;
+          pdc.publicDashboard.orgId = orgId;
+        }
         setPublicDashboardConfig(pdc);
       })
       .catch(() => {
         dispatch(notifyApp(createErrorNotification('Failed to retrieve public dashboard config')));
```

Comment:
this should not be needed, if the API returns an HTTP error error code with a message it should be shown in an alert box already

Reply:
This was checking to see if the PublicDashboardConfig.PublicDashboard is set rather than an http error. We populate this on the backend now.

Reply:
But why the catch then? the catch and nofifyApp should not be needed, if the API call fails it will already be shown?